tests/modes.spec.ts

```diff
@@ -141,3 +141,36 @@ test("testing add a route and save it", async () => {
     page.getByRole("button", { name: "1) Untitled route" })
   ).toBeVisible();
 });
+
+test("the viewport changes only once when opening a form", async () => {
+  let defaultViewport = new URL(page.url()).hash;
+
+  // Create a point, and make sure the viewport hasn't changed
+  await page.getByRole("button", { name: "New point" }).click();
+  await clickMap(page, 500, 500);
+  expect(new URL(page.url()).hash).toEqual(defaultViewport);
+
+  // Close the form
+  await page.getByRole("button", { name: "Save" }).click();
+
+  // Open the form from the sidebar. The viewport should change
+  await page.getByRole("button", { name: "1) Untitled point" }).click();
+  // The zoom effect is set to happen over 500ms, so wait a bit
```

Comment:
I'd love to make Playwright's auto-await magic work for all the maplibre interactions, but not sure how. Manual waits are the workaround.